docs/root/start/sandboxes/locality_load_balancing.rst

```diff
@@ -0,0 +1,141 @@
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+        Used to make ``HTTP`` requests.
+
+This example creates a setup for demonstrating the :ref:`locality weighted load balancing <arch_overview_load_balancing_locality_weighted_lb>` feature in Envoy proxy. The demo simulates a scenario that a backend service resides in two local zones and one remote zone.
+
+The components used in this demo are as follows:
+
+- A client container: runs Envoy proxy
+- Backend container in the same locality as the client, with priority set to 0, namely ``local-1``.
+- Backend container in the same locality as the client, with priority set to 1, namely ``local-2``.
+- Backend container in the the remote locality, with priority set to 1, namely ``remote-1``.
+- Backend container in the the remote locality, with priority set to 2, namely ``remote-2``.
+
+The client Envoy proxy configures the 4 backend containers in the same Envoy cluster, so that Envoy handles load balancing to those backend servers.
+
+Step 1: Start all of our containers
+***********************************
+
+In terminal, move to the ``examples/locality_load_balancing`` directory.
+
+To build this sandbox example and start the example services, run the following commands:
+
+.. code:: shell
+
+    # Start demo
+    $ docker-compose up --build -d
+
+The locality configuration is set in the client container via static Envoy configuration file. Please refer to the :download:`clusters configuration <_include/locality-load-balancing/envoy-client.yaml>` file.
+
+Step 2: Scenario with one replica in the highest priority locality
+******************************************************************
+
+In this scenario, each locality has 1 healthy replica running. In this case, all the requests should be sent to the locality with the highest priority, which is ``local-1``.
+
+.. code:: shell
+
+    # all requests to local-1
+    $ docker-compose exec -T client-envoy python3 client.py http://localhost:3000/ 100
+    Hello from backend-local-1!: 100, 100.0%
+    Failed: 0
+
+If locality ``local-1`` becomes unhealthy, the requests should be load balanced among the subsequent priority localities, which are ``local-2`` and ``remote-1``. They both have priority 1. We then send 100 requests to the backend cluster, and check the responders.
+
+.. code:: shell
+
+    # bring down local-1
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_1:8000/unhealthy
+    [backend-local-1] Set to unhealthy
+
+    # local-2 and remote-1 localities split the traffic 50:50
+    $ docker-compose exec -T client-envoy python3 client.py http://localhost:3000/ 100
+    Hello from backend-remote-1!: 51, 51.0%
+    Hello from backend-local-2!: 49, 49.0%
+    Failed: 0
+
+Now if ``local-2`` becomes unhealthy also, priority 1 locality is only 50% healthy. Requests will be sent to the ``remote-1`` and ``remote-2``.
```

Comment:
im struggling to understand this bit

what is the `priority 1 locality` ? Im guessing it is all of the configured localites where priority = `1`

its not immediately clear what triggers `remote-2` receiving the requests also when this didnt happen previously with `local-1` active

im guessing its because "priority 0" only has one configured backend - wondering how we can make this clearer

Reply:
1. yes, `priority 1 locality` means all the localities with priority = 1.
2. I think I need to mention before this line that request will be load balancer to lower priority localities if the healthiness of current locality drops to below 71% (by default). Therefore, the 50% healthiness activating the priority 2 locality becomes a clearer fact.